agent/host_agent_test.go

```diff
@@ -12,6 +12,7 @@ import (
 	"os"
 	"os/exec"
 	"runtime"
+	
```

Comment:
How about refactoring existing tests in this file to be run inside the container? That we can validate the helper function changes in this PR.